pkg/platform/runtime/setup/buildlog/artifactlog.go

```diff
@@ -0,0 +1,55 @@
+package buildlog
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+// ArtifactLog wraps a handler to listen for real-time build log messages for a specific artifact
+type ArtifactLog struct {
+	errCh chan error
+}
+
+// NewArtifactLog subscribes to events on the connection, and forwards build log events via the events handler
+func NewArtifactLog(artifactID artifact.ArtifactID, conn BuildLogConnector, events Events) (*ArtifactLog, error) {
```

Comment:
Can we make it a bit clearer from the structure of the code how this differs from `New()`? I'm grokking both as "a thing that logs artifact progress". Not clear to me what the distinction is without reading through all the code.

Reply:
I'll add a comment here explaining it:  We should not need it at all (https://github.com/ActiveState/cli/pull/1447#discussion_r670620444), because ideally the `BuildLog` can handle these messages on the websocket-connection that is given to it.  My hope is that once https://www.pivotaltracker.com/story/show/178901762 is resolved, that we can get rid of this struct completely...